src/PSR7/RequestValidator.php

```diff
@@ -102,4 +107,14 @@ private function findMatchingOperations(RequestInterface $request) : array
 
         return $pathFinder->search();
     }
+
+    /**
+     * Checks whether request matches any path in schema
+     */
+    private function hasMatchingPaths(RequestInterface $request) : bool
+    {
+        $pathFinder = new PathFinder($this->openApi, (string) $request->getUri(), $request->getMethod());
+
+        return $pathFinder->getPathMatches() !== [];
```

Comment:
Actually I think we can make it as fast caching data in `PathFinder` (it's statefyl at the moment, so we can do that easily). For this we need to.  Use same `PathFinder` twice